common/src/main/java/org/apache/celeborn/common/client/MasterClient.java

```diff
@@ -266,24 +266,13 @@ private RpcEndpointRef setupEndpointRef(String endpoint) {
   }
 
   private List<String> resolveMasterEndpoints() {
-    if (isWorker) {
+    if (isWorker && conf.internalPortEnabled()) {
       // For worker, we should use the internal endpoints if internal port is enabled.
-      if (conf.internalPortEnabled()) {
-        masterEndpointName = RpcNameConstants.MASTER_INTERNAL_EP;
-        return Arrays.asList(conf.masterInternalEndpoints());
-      } else {
-        masterEndpointName = RpcNameConstants.MASTER_EP;
-        return Arrays.asList(conf.masterEndpoints());
-      }
+      masterEndpointName = RpcNameConstants.MASTER_INTERNAL_EP;
+      return Arrays.asList(conf.masterInternalEndpoints());
     } else {
-      // This is for client, so we should use the secured endpoints if auth is enabled.
-      if (conf.authEnabled()) {
-        masterEndpointName = RpcNameConstants.MASTER_SECURED_EP;
-        return Arrays.asList(conf.masterSecuredEndpoints());
-      } else {
-        masterEndpointName = RpcNameConstants.MASTER_EP;
-        return Arrays.asList(conf.masterEndpoints());
-      }
+      masterEndpointName = RpcNameConstants.MASTER_EP;
```

Comment:
What's the behavior for client and internal port enable case? For this case, why not use internal endpoints? IMO, this would mislead the users to config internal port enable.

Reply:
Internal port is intended for servers to talk to each other. This is documented in the config. If your concern is that clients will connect to the Master on the internal port when auth is enabled, the still they will not be able to push/fetch data to the workers because they wouldn't have registered with the Master (which is part of the secure bootstrap).